sashimi/gui/camera_gui.py

```diff
@@ -47,237 +61,260 @@ class ViewingWidget(QWidget):
+        if self.auto_contrast:
+            self.frame_layer.reset_contrast_limits()
         else:
-            self.roi.visible = True
-            self.viewer.press_key(
-                "s"
-            )  # this allows moving the whole roi and scaling but no individual handles
+            self.set_manual_contrast()
 
-    def update_roi(self):
-        pass
+    def auto_contrast(self):
+        if self.auto_contrast_chk.isChecked():
+            self.wid_contrast_range.hide()
+            self.auto_contrast = True
+        else:
+            self.wid_contrast_range.show()
+            self.auto_contrast = False
+
+    def set_manual_contrast(self):
+        self.frame_layer.contrast_limits = [i for i in self.contrast_range.contrast_range]
 
 
-class CameraSettingsContainerWidget(QWidget):
+class CameraSettingsWidget(QWidget):
     """Widget to modify parameters for the camera.
 
     Parameters
     ----------
     state : State object
-    roi : default ROI size
+    wid_display : ViewingWidget
     timer : QTimer
     """
 
-    def __init__(self, state, roi, timer):
+    def __init__(self, state, wid_display, timer):
 
         super().__init__()
-        self.roi = roi
+        self.wid_display = wid_display
+        self.roi = wid_display.roi
+        self.roi_state = RoiState.FULL
         self.state = state
-        timer.timeout.connect(self.update_camera_info)
+        self.state.camera_settings.sig_param_changed.connect(self.update_on_bin_change)
+        self.timer = timer
+
+        self.camera_msg_queue = Queue()
+
+        self.sensor_resolution = self.wid_display.max_sensor_resolution
 
         self.full_size = True
+        self.current_binning = conf["camera"]["default_binning"]
 
         self.wid_camera_settings = ParameterGui(self.state.camera_settings)
 
         self.setLayout(QVBoxLayout())
 
-        self.lbl_camera_info = QLabel()
-        self.lbl_roi = QLabel()
-
-        self.set_roi_button = QPushButton("set ROI")
-        self.set_full_size_frame_button = QPushButton("set full size frame")
+        self.btn_roi = QPushButton(ROI_TEXTS[self.roi_state])
+        self.btn_cancel_roi = QPushButton("Cancel")
+        self.btn_cancel_roi.hide()
 
         self.layout().addWidget(self.wid_camera_settings)
-        self.layout().addWidget(self.lbl_camera_info)
-        self.layout().addWidget(self.set_roi_button)
-        self.layout().addWidget(self.set_full_size_frame_button)
-        self.layout().addWidget(self.lbl_roi)
-
-        self.update_camera_info()
-        self.set_full_size_frame()
-
-        self.set_roi_button.clicked.connect(self.set_roi)
-        self.set_full_size_frame_button.clicked.connect(self.set_full_size_frame)
+        self.layout().addWidget(self.btn_roi)
+        self.layout().addWidget(self.btn_cancel_roi)
+
+        self.btn_roi.clicked.connect(self.roi_action)
+        self.btn_cancel_roi.clicked.connect(self.cancel_roi_selection)
+
+    def roi_action(self):
```

Comment:
hmm this cycles through ROI states, maybe could be named better?

Reply:
and separated into two (iterate state and apply state change)

Reply:
not sure what the advantage would be in this case